src/script/calling/CallingRepository.ts

```diff
@@ -0,0 +1,971 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {Calling, GenericMessage} from '@wireapp/protocol-messaging';
+import {amplify} from 'amplify';
+import ko from 'knockout';
+import {t} from 'Util/LocalizerUtil';
+import {Logger, getLogger} from 'Util/Logger';
+import 'webrtc-adapter';
+import {Config} from '../auth/config';
+import {GENERIC_MESSAGE_TYPE} from '../cryptography/GenericMessageType';
+
+import {Environment} from 'Util/Environment';
+import {TIME_IN_MILLIS} from 'Util/TimeUtil';
+import {createRandomUuid} from 'Util/util';
+import {EventBuilder} from '../conversation/EventBuilder';
+import {MediaStreamHandler} from '../media/MediaStreamHandler';
+import {ModalsViewModel} from '../view_model/ModalsViewModel';
+import {TERMINATION_REASON} from './enum/TerminationReason';
+
+import {CALL_MESSAGE_TYPE} from './enum/CallMessageType';
+
+import {ConversationRepository} from '../conversation/ConversationRepository';
+import {EventInfoEntity} from '../conversation/EventInfoEntity';
+import {EventRepository} from '../event/EventRepository';
+import {MediaType} from '../media/MediaType';
+
+import {Call, ConversationId} from './Call';
+import {DeviceId, Participant, UserId} from './Participant';
+
+import {WebAppEvents} from '../event/WebApp';
+
+interface MediaStreamQuery {
+  audio?: boolean;
+  camera?: boolean;
+  screen?: boolean;
+}
+
+import {
+  CALL_TYPE,
+  CONV_TYPE,
+  ENV as AVS_ENV,
+  LOG_LEVEL,
+  REASON,
+  STATE as CALL_STATE,
+  VIDEO_STATE,
+  Wcall,
+  getAvsInstance,
+} from '@wireapp/avs';
+
+export class CallingRepository {
+  private readonly backendClient: any;
+  private readonly conversationRepository: any;
+  private readonly eventRepository: any;
+  private readonly mediaStreamHandler: MediaStreamHandler;
+  private readonly serverTimeHandler: any;
+
+  private selfUser: any;
+  private selfClientId: DeviceId;
+  private isReady: boolean = false;
+  private wUser: number | undefined;
+  private wCall: Wcall | undefined;
+  public readonly activeCalls: ko.ObservableArray<Call>;
+  private readonly isMuted: ko.Observable<boolean>;
+  private incomingCallCallback: Function;
+
+  // will cache the query to media stream (in order to avoid asking the system for streams multiple times when we have multiple peers)
+  private mediaStreamQuery: Promise<MediaStream> | undefined;
+
+  public readonly joinedCall: ko.PureComputed<Call | undefined>;
+
+  private callingConfig: any;
+  private callingConfigTimeout: number | undefined;
```

Comment:
```suggestion
  private callingConfigTimeout?: number;
```

Same goes for all other parameters which can be `undefined`.